main.go

```diff
@@ -2,49 +2,37 @@ package main
 
 import (
 	"encoding/json"
+	"errors"
 	"fmt"
 	"io"
 	"io/ioutil"
-	"log"
 	"net/http"
 	"os"
+	"path/filepath"
+	"strings"
 	"time"
 
-	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-steputils/stepconf"
+	"github.com/bitrise-io/go-utils/log"
 )
 
-const cacheArchivePath = "/tmp/cache-archive.tar"
-
-var (
-	gIsDebugMode = false
-)
-
-// StepParamsModel ...
-type StepParamsModel struct {
-	CacheAPIURL string
-	IsDebugMode bool
+// Config stores the step inputs.
+type Config struct {
+	CacheAPIURL string `env:"cache_api_url"`
+	DebugMode   bool   `env:"is_debug_mode,opt[true,false]"`
+	StackID     string `env:"BITRISE_STACK_ID"`
 }
 
-// GenerateDownloadURLRespModel ...
-type GenerateDownloadURLRespModel struct {
-	DownloadURL string `json:"download_url"`
-}
-
-// CreateStepParamsFromEnvs ...
-func CreateStepParamsFromEnvs() (StepParamsModel, error) {
-	stepParams := StepParamsModel{
-		CacheAPIURL: os.Getenv("cache_api_url"),
-		IsDebugMode: os.Getenv("is_debug_mode") == "true",
+// downloadCacheArchive downloads the cache archive and returns the downloaded file's path.
+// If the URI points to a local file it returns the local paths.
+func downloadCacheArchive(url string) (string, error) {
+	if strings.HasPrefix(url, "file://") {
+		return strings.TrimPrefix(url, "file://"), nil
 	}
 
-	return stepParams, nil
-}
-
-func downloadCacheArchive(url string) error {
-	// Get the data
 	resp, err := http.Get(url)
 	if err != nil {
-		return err
+		return "", err
 	}
 
 	defer func() {
```

Comment:
I think "" [!] Failed to close Archive download response body: %s", err)" should be a waring.